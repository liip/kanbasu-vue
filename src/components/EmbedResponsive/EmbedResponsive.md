```vue
<embed-responsive>
  <iframe src="https://www.youtube.com/embed/eHSSNyIlAps"></iframe>
</embed-responsive>
```

### With custom ratio

```vue
<embed-responsive :ratio="4/3">
  <iframe src="https://www.youtube.com/embed/0daS_SDCT_U"></iframe>
</embed-responsive>
```