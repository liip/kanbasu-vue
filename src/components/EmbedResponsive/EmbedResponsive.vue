<template>
  <div class="embed-responsive" :style="styleObject">
    <slot></slot>
  </div>
</template>

<script>
export default {
  name: 'embed-responsive',

  props: {
    ratio: {
      type: Number,
      default: 16 / 9,
    },
  },

  computed: {
    styleObject() {
      return {
        paddingTop: `${(1 / this.ratio) * 100}%`,
      };
    },
  },
};
</script>
