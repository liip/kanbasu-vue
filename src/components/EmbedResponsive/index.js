import EmbedResponsive from './EmbedResponsive.vue';

EmbedResponsive.install = function(Vue) {
  Vue.component(EmbedResponsive.name, EmbedResponsive);
};

export default EmbedResponsive;
