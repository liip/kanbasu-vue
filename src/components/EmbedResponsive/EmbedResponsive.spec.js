import { shallowMount } from '@vue/test-utils';
import EmbedResponsive from './EmbedResponsive.vue';

describe('EmbedResponsive.vue', () => {
  it('renders', () => {
    const wrapper = shallowMount(EmbedResponsive, {
      slots: {
        default: '<iframe src="#"></iframe>',
      },
    });
    expect(wrapper).toMatchSnapshot();
  });
});

describe('EmbedResponsive.vue', () => {
  it('renders with custom ratio', () => {
    const wrapper = shallowMount(EmbedResponsive, {
      slots: {
        default: '<iframe src="#"></iframe>',
      },
      propsData: {
        ratio: 4 / 3,
      },
    });
    expect(wrapper).toMatchSnapshot();
  });
});
