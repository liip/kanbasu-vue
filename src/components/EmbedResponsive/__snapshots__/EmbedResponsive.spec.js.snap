// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmbedResponsive.vue renders 1`] = `<div class="embed-responsive" style="padding-top: 56.25%;"><iframe src="#"></iframe></div>`;

exports[`EmbedResponsive.vue renders with custom ratio 1`] = `<div class="embed-responsive" style="padding-top: 75%;"><iframe src="#"></iframe></div>`;
